subsys/bluetooth/services/rscs.c

```diff
@@ -0,0 +1,442 @@
+		ctrl_point_ind(conn, req->op, SC_CP_RSP_OP_NOT_SUPP, NULL, 0);
+		return len;
+	}
+
+	if (!ctrl_point_param_validate(req, len)) {
+		LOG_ERR("Invalid parameter for op code %#x", req->op);
+		ctrl_point_ind(conn, req->op, SC_CP_RSP_INVAL_PARAM, NULL, 0);
+		return len;
+	}
+
+	switch (req->op) {
+	case SC_CP_OP_SET_CUMULATIVE:
+		if (rscs_inst.cp_cb && rscs_inst.cp_cb->set_cumulative) {
+			if (!rscs_inst.cp_cb->set_cumulative(sys_le32_to_cpu(req->total_dist))) {
+				status = SC_CP_RSP_SUCCESS;
+				break;
+			}
+		}
+		status = SC_CP_RSP_FAILED;
+		break;
+	case SC_CP_OP_CALIBRATION:
+		if (rscs_inst.cp_cb && rscs_inst.cp_cb->calibrarion) {
+			if (!rscs_inst.cp_cb->calibrarion()) {
+				status = SC_CP_RSP_SUCCESS;
+				break;
+			}
+		}
+		status = SC_CP_RSP_FAILED;
+		break;
+	case SC_CP_OP_UPDATE_LOC:
+		status = SC_CP_RSP_INVAL_PARAM;
+		for (size_t i = 0; i < rscs_inst.num_location; i++) {
+			if (rscs_inst.supported_locations[i] == req->location) {
+				rscs_inst.sensor_loc = req->location;
+				status = SC_CP_RSP_SUCCESS;
+				break;
+			}
+		}
+
+		if (status == SC_CP_RSP_SUCCESS && rscs_inst.cp_cb && rscs_inst.cp_cb->update_loc) {
+			rscs_inst.cp_cb->update_loc(rscs_inst.sensor_loc);
+		}
+		break;
+	case SC_CP_OP_REQ_SUPP_LOC:
+		ctrl_point_ind(conn, req->op, SC_CP_RSP_SUCCESS, &rscs_inst.supported_locations,
+			       rscs_inst.num_location);
+		return len;
+	default:
+		status = SC_CP_RSP_OP_NOT_SUPP;
+		break;
+	}
+
+	ctrl_point_ind(conn, req->op, status, NULL, 0);
+
+	return len;
+}
+
+BT_GATT_SERVICE_DEFINE(rsc_svc,
+	BT_GATT_PRIMARY_SERVICE(BT_UUID_RSCS),
+	BT_GATT_CHARACTERISTIC(BT_UUID_RSC_MEASUREMENT, BT_GATT_CHRC_NOTIFY,
+			       BT_GATT_PERM_NONE, NULL, NULL, NULL),
+	BT_GATT_CCC(rsc_meas_ccc_cfg_changed, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
+	BT_GATT_CHARACTERISTIC(BT_UUID_RSC_FEATURE, BT_GATT_CHRC_READ,
+			       BT_GATT_PERM_READ, read_rsc_featute, NULL, NULL),
+	BT_GATT_CHARACTERISTIC(BT_UUID_SENSOR_LOCATION, BT_GATT_CHRC_READ,
+			       BT_GATT_PERM_READ, read_location, NULL, &rscs_inst.sensor_loc),
+	BT_GATT_CHARACTERISTIC(BT_UUID_SC_CONTROL_POINT, BT_GATT_CHRC_WRITE | BT_GATT_CHRC_INDICATE,
+			       BT_GATT_PERM_WRITE, NULL, write_ctrl_point, &rscs_inst.sensor_loc),
+	BT_GATT_CCC(ctrl_point_ccc_cfg_changed, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
+);
+
+static void indicate_cb(struct bt_conn *conn,
+			struct bt_gatt_indicate_params *params, uint8_t err)
+{
+	LOG_DBG("Indication %s", err != 0U ? "fail" : "success");
+}
+
+static void indicate_destroy(struct bt_gatt_indicate_params *params)
+{
+	LOG_DBG("Indication complete");
+	rscs_inst.indicating = false;
+}
+
+static void ctrl_point_ind(struct bt_conn *conn, uint8_t req_op, uint8_t status,
+			   const void *data, uint16_t len)
+{
+	struct sc_ctrl_point_ind *ind;
+	NET_BUF_SIMPLE_DEFINE(buf, sizeof(*ind) + len);
+
+	if (rscs_inst.indicating) {
+		return;
+	}
+
+	buf.len = buf.size;
+	ind = (void *)buf.data;
```

Comment:
Please use net_buf API.
Here for example:
`ind = net_buf_simple_add(buf, (sizeof(*ind) + len))`

Reply:
The same for len and other function which uses net buf

Reply:
Fixed.